### PR TITLE
Added authentication flow and action queues

### PR DESCRIPTION
**Pull request by nraomobiquityinc**
1. Added Slacker authentication flow
2. Added example authenticated action for deleting a user's files

### CODE REVIEW

**Comment by nraomobiquityinc** [comment]
@Ustice Once we've reviewed this, I'll update the README with details about starting redis etc.

File: actions/deleteFiles.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+  setup: function() {
+    // This method will be run at server start up.
+  },
+
+  execute: function(data, callback) {
+    var accessToken = data.accessToken;
+    listFiles(deleteFile);
+
+    function listFiles(deleteFile) {
```
Comment (Ustice) [inline actions/deleteFiles.js]: Move outside of the action.

File: authenticationHelper.js [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+              "\nYou only need to do this once. Your command will be run once you've authorized us");
+            return true;
+          }
+        });
+      } else {
+        //user has tried to issue an authenticated action in the past but has never authorized us
+        //enqueue this action and give him another chance to authorize us
+        pastQueuedActionsList = JSON.parse(pastQueuedActionsList);
+        pastQueuedActionsList.push(actionData);
+        redisClient.hset(data.user_id, "queuedActions", JSON.stringify(pastQueuedActionsList), function(err, res) {
```
Comment (Ustice) [inline authenticationHelper.js:79]: Wrap reddis calls to keep code DRY.

File: authenticationHelper.js [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+            var action = actionsData[idx];
+            var command = _.clone(action.data.command); //これは必要かないかまだはっきり分からない
+            var data = action.data;
+            bot.processCommand(command, data, response, function(message) {
+              bot.sendMessage(message + "\n*Action command*: `" + command.name + "`" +
+                "\n*Action initially requested at*: " + action.timeStamp, userId);
+            });
+          })
+        }
+      })
```
Comment (Ustice) [inline authenticationHelper.js]: semicolon

**Comment by nraomobiquityinc** [comment]
Going to close this request and make a pull to another branch so we don't pollute this timeline.

**Pull request closed without merging**